source/extensions/filters/http/aws_lambda/sts_connection_pool.cc

```diff
@@ -114,64 +128,77 @@ StsConnectionPoolImpl::add(StsConnectionPool::Context::Callbacks *callbacks) {
   return ctx_ptr;
 };
 
+void StsConnectionPoolImpl::addChained( std::string role_arn) {
+  chained_requests_.emplace_back(role_arn);
+  return ;
+};
+
+void StsConnectionPoolImpl::markFailed( CredentialsFailureStatus status) {
+  onFailure(status);
+};
+
 void StsConnectionPoolImpl::onSuccess(const absl::string_view body) {
   ASSERT(!body.empty());
+  
   request_in_flight_ = false;
 
-// using a macro as we need to return on error
-// TODO(yuval-k): we can use string_view instead of string when we upgrade to
-// newer absl.
-#define GET_PARAM(X)                                                           \
-  std::string X;                                                               \
-  {                                                                            \
-    std::match_results<absl::string_view::const_iterator> matched;             \
-    bool result = std::regex_search(body.begin(), body.end(), matched,         \
-                                    StsResponseRegex::get().regex_##X);        \
-    if (!result || !(matched.size() != 1)) {                                   \
-      ENVOY_LOG(trace, "response body did not contain " #X);                   \
-      onFailure(CredentialsFailureStatus::InvalidSts);                         \
-      return;                                                                  \
-    }                                                                          \
-    const auto &sub_match = matched[1];                                        \
-    decltype(X) matched_sv(sub_match.first, sub_match.length());               \
-    X = std::move(matched_sv);                                                 \
-  }
+  // using a macro as we need to return on error
```

Comment:
its hard to see what changed here with all the white spaces in the diff; can we revert changes that are just whitespace?

Reply:
Sure thing. I have reverted white space changes throughout.